gcloud/storage/batch.py

```diff
@@ -170,10 +170,22 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 def _unpack_batch_response(response, content):
     """Convert response, content -> [(status, reason, payload)]."""
     parser = Parser()
-    faux_message = ('Content-Type: %s\nMIME-Version: 1.0\n\n%s' %
-                    (response['content-type'], content))
-
-    message = parser.parsestr(faux_message)
+    if not isinstance(content, six.binary_type):
```

Comment:
Well the reason we use bytes is so they can be `b''` joined; it fixed the `%s` issue with bytes in Py3.

The native `str` for email part happens below at `six.PY2`.

I can add a comment here explaining we coerce to bytes so the concat is valid in Py3. WDYT?